### PR TITLE
feat: add add-to-cart component

### PR DESCRIPTION
**Pull request by negibsherif**

### CODE REVIEW

File: add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import { Bindings, resultContext, initializeBindings } from "@coveo/atomic";
+import { Component, Host, h, Element, State, forceUpdate } from "@stencil/core";
+import { Result, Unsubscribe } from "@coveo/headless";
+
+interface CustomResult extends Result {
+  priceAndAvailability:
```
Comment (jantoun-coveo) [inline add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx]: This needs to be removed it is a client specific which render the component not generic

File: add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    this.i18nUnsubscribe = () =>
+      this.bindings!.i18n.off("languageChanged", updateLanguage);
+
+  }
+
+  private onAddToOptionClicked = async (e: Event) => {
+    //Callback function to add element to cart
+  };
+
+  changeQuantity(e: Event) {
```
Comment (jantoun-coveo) [inline add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx]: client specific we don't need it

File: add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    const qtyValue = parseFloat((e.target as HTMLInputElement).value);
+
+    this.quantity = qtyValue;
+  }
+
+  checkIsNumber(e: KeyboardEvent) {
```
Comment (jantoun-coveo) [inline add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx]: we don't need this function

File: add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    if ((this.quantity?.toString()?.length || 0) + 1 > 11) {
+      e.preventDefault();
+      return false;
+    }
+    return true;
+  }
+
+  updateQuantity(e: KeyboardEvent) {
```
Comment (jantoun-coveo) [inline add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx]: we don't need this function

File: add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    this.quantity = parseFloat((e.target as HTMLInputElement).value);
+  }
+
+  public disconnectedCallback() {
+    this.i18nUnsubscribe();
+  }
+
+  render() {
+    const isAvailable = true;
```
Comment (jantoun-coveo) [inline add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx]: we don't need this

File: add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    const isPublished = true;
+
+    if (!isAvailable || !isPublished) {
```
Comment (jantoun-coveo) [inline add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx]: We don't need this

File: add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+      this.host.remove();
+      return;
+    }
+    return (
+      <Host>
+        {isAvailable ? (
```
Comment (jantoun-coveo) [inline add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx]: We don't need this

File: add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+          <div
+            class="add-to-cart_wrapper"
+          >
+            <div class="add-to-cart_wrapper__units">
+              <input
```
Comment (jantoun-coveo) [inline add-to-cart/src/components/add-to-cart/src/add-to-cart.tsx]: We don't need the input this is just a classic button that will trigger an add to cart action

File: add-to-cart/package.json [hunk 9]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "@coveo/sample-component-project",
```
Comment (jantoun-coveo) [inline add-to-cart/package.json:2]: Was the project created as a default sample to have this name ?

File: add-to-cart/src/components/add-to-cart/src/add-to-cart.css [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+.add-to-cart_wrapper__units {
+  display: flex;
+  justify-content: flex-end;
+  height: 30px;
+  width: 100%;
+}
+
+.add-to-cart_wrapper__units input {
```
Comment (jantoun-coveo) [inline add-to-cart/src/components/add-to-cart/src/add-to-cart.css]: make sure that the css meets the components after it was refactored

**Review by jantoun-coveo — CHANGES_REQUESTED** [review]
a lot of work is needed for this one, you need to not just copy paste there is a lot of stuff that are client specific and needs to be removed and changed.

File: add-to-cart/package.json (hunk 9 above)
Comment (negibsherif) [reply add-to-cart/package.json:2]: Locally this is different. Not sure what happened, never seen this happen before that's weird. Even git lens shows that in the previous commit made two days ago, this text is different. Anyways, I'll check to make sure this is fixed on the next commit.  
Screenshot 2023-03-08 at 11 16 01 AM

Comment (negibsherif) [reply add-to-cart/package.json:2]: @jantoun-coveo actually yeah there is another package.json file in the components/add-to-cart/src folder and that's the one with the screenshot I sent. This was the command I used to create the component. `npm init @coveo/atomic-component add-to-cart`

**Comment by negibsherif** [comment]
@jantoun-coveo This is ready for review again.

**Comment by negibsherif** [comment]
Note: the read me has been updated, once you approve of the read me for this component, I can add something similar to the rest of the components as well.

**Comment by negibsherif** [comment]
Making a change with the callback function

**Review by jantoun-coveo — APPROVED** [review]

**Pull request merged**